packages/autocertifier-client/src/RestClient.ts

```diff
@@ -0,0 +1,108 @@
+import { Session } from './data/Session'
+import { CertifiedSubdomain } from './data/CertifiedSubdomain'
+import request from 'request'
+import { UpdateIpAndPortRequest } from './data/UpdateIpAndPortRequest'
+import { CreateCertifiedSubdomainRequest } from './data/CreateCertifiedSubdomainRequest'
+import { ServerError } from './errors'
+import { Logger } from '@streamr/utils'
+
+const logger = new Logger(module)
+
+// TODO: use a non-deprecated HTTP client that support async/await instead of request
+export class RestClient {
+
+    private baseUrl: string
+
+    constructor(baseUrl: string) {
+        this.baseUrl = baseUrl
+    }
+
+    public async createSession(): Promise<string> {
+        const url = this.baseUrl + '/sessions'
+        try {
+            const response = await this.post<Session>(url, {})
+            return response.sessionId
+        } catch (err) {
+            logger.debug(err)
+            throw err
+        }
+    }
+
+    public async createNewSubdomainAndCertificate(streamrWebSocketPort: number, sessionId: string): Promise<CertifiedSubdomain> {
+        const url = this.baseUrl + '/certifiedSubdomains'
```

Comment:
Could unify the path name (now we have `/certifiedSubdomains` here and `/certifiedsubdomains` at line 42)
```suggestion
        const url = this.baseUrl + '/certified-subdomains'
```